src/muse/sectors/sector.py

```diff
@@ -151,6 +151,8 @@ def __init__(
         It can be anything registered with
         :py:func:`@register_production<muse.production.register_production>`.
         """
+        self.output_data: xr.Dataset | None = None
```

Comment:
Are you actually using this property before an `xr.DataSet` is assigned to it? If not, you could get away with this.

```suggestion
        self.output_data: xr.Dataset
```